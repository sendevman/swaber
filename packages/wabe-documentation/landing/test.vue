
<script setup lang="ts">
import './tailwind.css'

import { ref, computed } from 'vue'

const darkMode = ref(false) // or use your theme management logic

const themeClass = computed(() => (darkMode.value ? 'dark' : ''))
</script>

<template>
  <div :class="themeClass">
    <!-- Hero Section -->
    <section class="bg-gradient-to-r from-blue-900 to-blue-600 text-white py-16">
      <div class="container mx-auto px-4 py-16 flex flex-col items-center text-center">
        <h1 class="text-4xl font-bold mb-4">Empower Your Development with Wabe</h1>
        <p class="text-lg mb-8">The open-source backend-as-a-service solution built with TypeScript. Fast, secure, and extensible.</p>
        <a href="#get-started" class="bg-blue-600 text-white py-2 px-4 rounded-lg font-semibold shadow-lg hover:bg-blue-700 transition-colors">Get Started</a>
      </div>
    </section>

    <!-- Features Section -->
    <section class="bg-gray-100 dark:bg-gray-900 py-16">
      <div class="container mx-auto px-4">
        <h2 class="text-3xl font-semibold text-center mb-8">Features</h2>
        <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8">
          <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Authentication</h3>
            <p>Seamless authentication with email, password, OAuth, and custom methods.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">GraphQL API</h3>
            <p>Powerful and flexible GraphQL API to interact with your data.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Permissions Security</h3>
            <p>Granular permissions to secure your data and control access.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Hooks System</h3>
            <p>Before and after request hooks for fine-tuned control over your database interactions.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">File Upload</h3>
            <p>Easy file uploads integrated into your backend service.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Dashboard & Integrations</h3>
            <p>Comprehensive dashboard and upcoming Mailgun & Stripe integrations.</p>
          </div>
        </div>
      </div>
    </section>

    <!-- Future Features Section -->
    <section class="bg-gray-200 dark:bg-gray-800 py-16">
      <div class="container mx-auto px-4">
        <h2 class="text-3xl font-semibold text-center mb-8">Coming Soon</h2>
        <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8">
          <div class="p-6 bg-white dark:bg-gray-900 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Mailgun Integration</h3>
            <p>Effortlessly integrate with Mailgun to handle your email services.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-900 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Stripe Integration</h3>
            <p>Seamless payment processing with Stripe.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-900 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Generated REST API</h3>
            <p>Automatic REST API generation for added flexibility.</p>
          </div>
          <div class="p-6 bg-white dark:bg-gray-900 rounded-lg shadow-lg border border-gray-200 dark:border-gray-700">
            <h3 class="text-xl font-semibold mb-4">Enhanced Dashboard</h3>
            <p>Advanced analytics and management tools in the dashboard.</p>
          </div>
        </div>
      </div>
    </section>

    <!-- Get Started Section -->
    <section class="bg-gray-900 text-white py-16">
      <div class="container mx-auto px-4 text-center">
        <h2 class="text-3xl font-bold mb-4">Get Started with Wabe</h2>
        <p class="text-lg mb-8">Ready to integrate Wabe into your project? Use the following command:</p>
        <code class="bg-gray-800 text-white py-2 px-4 rounded-lg font-mono text-xl">bun add wabe</code>
      </div>
    </section>
  </div>
</template>

