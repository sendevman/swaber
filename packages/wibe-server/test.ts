import { gql } from 'graphql-request'

const run = async () => {
    const res = await fetch('http://localhost:3000/graphql', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({
            query: gql`
                query IntrospectionSchema {
                    __schema {
                        types {
                            name
                            description
                            kind
                            inputFields {
                                name
                                type {
                                    name
                                    kind
                                    ofType {
                                        name
                                        kind
                                    }
                                }
                            }
                            fields {
                                name
                                description
                                args {
                                    name
                                    description
                                    type {
                                        name
                                        kind
                                        ofType {
                                            name
                                            kind
                                        }
                                    }
                                }
                            }
                        }
                        mutationType {
                            name
                            description
                            fields {
                                name
                                description
                                args {
                                    name
                                    description
                                    type {
                                        name
                                        kind
                                        ofType {
                                            name
                                            kind
                                        }
                                    }
                                }
                                type {
                                    name
                                    kind
                                    ofType {
                                        name
                                        kind
                                    }
                                }
                            }
                        }
                        queryType {
                            name
                            description
                            fields {
                                name
                                description
                                args {
                                    name
                                    description
                                    type {
                                        name
                                        kind
                                        ofType {
                                            name
                                            kind
                                        }
                                    }
                                }
                                type {
                                    name
                                    kind
                                    ofType {
                                        name
                                        kind
                                    }
                                }
                            }
                        }
                    }
                }
            `,
        }),
    })

    const {
        data: {
            __schema: { queryType, mutationType, types: objectTypes },
        },
    } = await res.json()

    const types = objectTypes.map((type: any) => {
        const fields = type.fields?.map((field: any) => {
            const args = field.args.map((arg: any) => {
                return `${arg.name}: ${arg.type.name}`
            })
        })

        if (type.kind === 'SCALAR' || type.kind === 'ENUM') {
            return `${type.name}`
        }

        if (type.kind === 'INPUT_OBJECT') {
            const fields = type.inputFields.map((field: any) => {
                return `${field.name}: ${field.type.name}`
            })

            return `${type.name} { ${fields.join(', ')} }`
        }

        return `${type.name} { ${fields.join(', ')} }`
    })

    console.log(types)

    const queries = queryType.fields.map((field: any) => {
        const args = field.args.map((arg: any) => {
            return `${arg.name}: ${arg.type.name}`
        })

        return `${field.name}(${args.join(', ')}): ${field.type.name === null ? field.type.ofType.name : field.type.name}${field.type.kind === 'NON_NULL' ? '!' : ''}`
    })

    const mutations = mutationType.fields.map((field: any) => {
        const args = field.args.map((arg: any) => {
            return `${arg.name}: ${arg.type.name}`
        })

        return `${field.name}(${args.join(', ')}): ${field.type.name === null ? field.type.ofType.name : field.type.name}${field.type.kind === 'NON_NULL' ? '!' : ''}`
    })
}

run().catch((err) => {
    console.error(err)
})
