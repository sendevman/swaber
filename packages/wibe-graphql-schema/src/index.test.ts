import { describe, expect, it, mock } from 'bun:test'
import { getGraphqlSchema } from './index'

const mockFetch = mock(() => ({
    json: async () => ({
        data: {
            __schema: {
                types: [
                    {
                        name: 'Phone',
                        kind: 'SCALAR',
                    },
                    {
                        name: 'Email',
                        kind: 'SCALAR',
                    },
                    {
                        name: 'Role',
                        kind: 'ENUM',
                        enumValues: [
                            {
                                name: 'Admin',
                            },
                            { name: 'User' },
                        ],
                    },
                    {
                        name: 'Provider',
                        kind: 'ENUM',
                        enumValues: [
                            {
                                name: 'Google',
                            },
                            { name: 'X' },
                        ],
                    },
                    {
                        name: 'Query',
                        kind: 'OBJECT',
                        fields: [
                            {
                                name: 'findOneUser',
                                description: 'Get a user by id',
                                type: {
                                    name: 'User',
                                    kind: 'OBJECT',
                                },
                                args: [
                                    {
                                        name: 'id',
                                        description: 'The id of the user',
                                        type: {
                                            name: 'ID',
                                            kind: 'SCALAR',
                                        },
                                    },
                                ],
                            },
                            {
                                name: 'findManyUser',
                                description: 'Get a list of users',
                                type: {
                                    name: null,
                                    kind: 'LIST',
                                    ofType: {
                                        name: null,
                                        kind: 'NON_NULL',
                                        ofType: {
                                            name: 'UserConnection',
                                        },
                                    },
                                },
                                args: [
                                    {
                                        name: 'where',
                                        description: 'The where clause',
                                        type: {
                                            name: 'UserWhereInput',
                                            kind: 'INPUT_OBJECT',
                                        },
                                    },
                                ],
                            },
                        ],
                    },
                    {
                        name: 'Mutation',
                        kind: 'OBJECT',
                        fields: [
                            {
                                name: 'createOneUser',
                                description: 'Create a user',
                                type: {
                                    name: 'User',
                                    kind: 'OBJECT',
                                },
                                args: [
                                    {
                                        name: 'age',
                                        description: 'The age of the user',
                                        type: {
                                            name: 'Int',
                                            kind: 'SCALAR',
                                        },
                                    },
                                    {
                                        name: 'name',
                                        description: 'The name of the user',
                                        type: {
                                            name: 'String',
                                            kind: 'SCALAR',
                                        },
                                    },
                                ],
                            },
                            {
                                name: 'createManyUser',
                                description: 'Create a list of users',
                                type: {
                                    name: null,
                                    kind: 'NON_NULL',
                                    ofType: {
                                        name: null,
                                        kind: 'LIST',
                                        ofType: {
                                            name: null,
                                            kind: 'NON_NULL',
                                            ofType: {
                                                name: 'User',
                                            },
                                        },
                                    },
                                },
                                args: [
                                    {
                                        name: 'name',
                                        description: 'The name of the user',
                                        type: {
                                            name: 'String',
                                            kind: 'SCALAR',
                                        },
                                    },
                                    {
                                        name: 'age',
                                        description: 'The age of the user',
                                        type: {
                                            name: 'Int',
                                            kind: 'SCALAR',
                                        },
                                    },
                                ],
                            },
                            {
                                name: 'createManyUserWithOnlyTypeRequired',
                                description: 'Create a list of users',
                                type: {
                                    name: null,
                                    kind: 'NON_NULL',
                                    ofType: {
                                        name: 'User',
                                    },
                                },
                                args: [
                                    {
                                        name: 'name',
                                        description: 'The name of the user',
                                        type: {
                                            name: 'String',
                                            kind: 'SCALAR',
                                        },
                                    },
                                    {
                                        name: 'age',
                                        description: 'The age of the user',
                                        type: {
                                            name: 'Int',
                                            kind: 'SCALAR',
                                        },
                                    },
                                ],
                            },
                        ],
                    },
                    {
                        name: 'User',
                        kind: 'OBJECT',
                        fields: [
                            {
                                name: 'id',
                                description: 'The id of the user',
                                type: {
                                    name: 'ID',
                                    kind: 'SCALAR',
                                },
                            },
                            {
                                name: 'name',
                                description: 'The name of the user',
                                type: {
                                    name: 'String',
                                    kind: 'SCALAR',
                                },
                            },
                        ],
                    },
                    {
                        name: 'UserConnection',
                        kind: 'OBJECT',
                        fields: [
                            {
                                name: 'edges',
                                description: 'The edges of the user',
                                type: {
                                    name: 'UserEdge',
                                    kind: 'OBJECT',
                                },
                            },
                        ],
                    },
                    {
                        name: 'UserWhereInput',
                        kind: 'INPUT_OBJECT',
                        inputFields: [
                            {
                                name: 'id',
                                description: 'The id of the user',
                                type: {
                                    name: 'ID',
                                    kind: 'SCALAR',
                                },
                            },
                            {
                                name: 'name',
                                description: 'The name of the user',
                                type: {
                                    name: 'String',
                                    kind: 'SCALAR',
                                },
                            },
                        ],
                    },
                ],
            },
        },
    }),
}))
// @ts-expect-error
global.fetch = mockFetch

describe('GetGraphqlSchema', () => {
    it('should return a valid json graphql schema', async () => {
        const { enums, scalars, queries, mutations } = await getGraphqlSchema(
            'http://localhost:3000/graphql',
        )

        const schemas = [
            enums.join('\n'),
            scalars.join('\n'),
            queries,
            mutations,
        ].flat()

        expect(schemas).toMatchSnapshot()
    })
})
