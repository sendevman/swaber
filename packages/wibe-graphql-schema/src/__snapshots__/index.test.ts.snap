// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`GetGraphqlSchema should return a valid json graphql schema 1`] = `
[
  
"enum Role {
	Admin,
	User
}"
,
  
"enum Provider {
	Google,
	X
}"
,
  "scalar Phone",
  "scalar Email",
  
"type Query {
	"""
	Get a user by id
	"""
	findOneUser(id: ID): User
	"""
	Get a list of users
	"""
	findManyUser(where: UserWhereInput): [UserConnection!]
}"
,
  
"type Mutation {
	"""
	Create a user
	"""
	createOneUser(age: Int, name: String, nonNullArrayOfNonNullType: [String!]!): User
	"""
	Create a list of users
	"""
	createManyUser(name: String, age: Int): [User!]!
	"""
	Create a list of users
	"""
	createManyUserWithOnlyTypeRequired(name: String, age: Int): User!
}"
,
  
"type Query {
	findOneUser: User
	findManyUser: [UserConnection!]
}
type Mutation {
	createOneUser: User
	createManyUser: [User!]!
	createManyUserWithOnlyTypeRequired: User!
}
type User {
	id: ID
	name: String
}
type UserConnection {
	edges: UserEdge
}"
,
  
"input UserWhereInput {
	id: ID
	name: String
	nonNullListOfNonNullInput: [NonNullField!]!
}"
,
]
`;
